tool.c

```diff
@@ -221,9 +222,105 @@ compdb(int argc, char *argv[])
 	return 0;
 }
 
+static void
+targets_rule(int argc, char *argv[])
+{
+	struct edge *e = NULL;
+	for (e = alledges; e; e = e->allnext) {
+		if (argc == 2) {
+			if (e->nin == 0)
+				continue;
+
+			if (!e->in[0]->gen)
+				printf("%s\n", e->in[0]->path->s);
+		}
+		else if (argc == 3) {
```

Comment:
This should be merged with the previous line with the closing `}`.

Reply:
Done